nessie/src/main/java/org/apache/iceberg/nessie/NessieViewOperations.java

```diff
@@ -0,0 +1,129 @@
+
+public class NessieViewOperations extends BaseViewOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(NessieViewOperations.class);
+
+  private final NessieIcebergClient client;
+  private final ContentKey key;
+  private final FileIO fileIO;
+  private IcebergView icebergView;
+
+  NessieViewOperations(ContentKey key, NessieIcebergClient client, FileIO fileIO) {
+    this.key = key;
+    this.client = client;
+    this.fileIO = fileIO;
+  }
+
+  @Override
+  public void doRefresh() {
+    try {
+      client.refresh();
+    } catch (NessieNotFoundException e) {
+      throw new RuntimeException(
+          String.format(
+              "Failed to refresh as ref '%s' is no longer valid.", client.getRef().getName()),
+          e);
+    }
+    String metadataLocation = null;
+    Reference reference = client.getRef().getReference();
+    try {
+      Content content = client.getApi().getContent().key(key).reference(reference).get().get(key);
+      LOG.debug("Content '{}' at '{}': {}", key, reference, content);
+      if (content == null) {
+        if (currentMetadataLocation() != null) {
+          throw new NoSuchViewException("View does not exist: %s in %s", key, reference);
+        }
+      } else {
+        this.icebergView =
+            content
+                .unwrap(IcebergView.class)
+                .orElseThrow(() -> new NessieContentNotFoundException(key, reference.getName()));
+        metadataLocation = icebergView.getMetadataLocation();
+      }
+    } catch (NessieNotFoundException ex) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchViewException("View does not exist: %s in %s", key, reference);
+      }
+    }
+    refreshFromMetadataLocation(
+        metadataLocation,
+        null,
+        2,
+        location ->
+            NessieUtil.loadViewMetadata(
+                ViewMetadataParser.read(io().newInputFile(location)), location, reference));
+  }
+
+  @Override
+  public void doCommit(ViewMetadata base, ViewMetadata metadata) {
+    String newMetadataLocation = writeNewMetadataIfRequired(metadata);
+
+    boolean failure = false;
+    try {
+      String contentId = icebergView == null ? null : icebergView.getId();
+      client.commitView(base, metadata, newMetadataLocation, contentId, key);
+    } catch (NessieConflictException | NessieNotFoundException | HttpClientException ex) {
+      if (ex instanceof NessieConflictException || ex instanceof NessieNotFoundException) {
+        failure = true;
+      }
+      NessieUtil.handleExceptionsForCommits(ex, client.refName(), Content.Type.ICEBERG_VIEW)
+          .ifPresent(
+              exception -> {
+                throw exception;
+              });
+    } catch (NessieBadRequestException ex) {
+      failure = true;
+      throw NessieUtil.handleBadRequestForCommit(client, key, Content.Type.ICEBERG_VIEW).orElse(ex);
+    } finally {
+      if (failure) {
+        io().deleteFile(newMetadataLocation);
```

Comment:
This code appears to be shared with `NessieTableOperations`. Could we refactor it to have the exception/failure/cleanup logic in one place?

Reply:
I tried before. 
But one works on `client.commitView` and one is `client.commitTable`. So, it cannot be extracted to common code.
I did also try accepting a generic Runnable to execute. But it cannot throw the required exception. Hence, it has to be duplicated.

Reply:
We could define our own functional `interface` with declared exceptions.